Contributing revamp

whatyouhide opened the pull request:
Made the changes described in #451. I added a `Gemfile` (with its `Gemfile.lock`) and updated the README with a more detailed guide on how to contribute to the website and how to use `bundle install/exec` now that a `Gemfile` is there.

**PS** hey @eksperimental, I'd appreciate if you gave a look at this too since you opened the issue in the first place :). Let me know what you think!

eksperimental commented:
thank for you the changes @whatyouhide 

I think we need to change a few things, I will list them here as a regular comment, as going in the code would be too complicated:

Provided the amount of steps needed, it think it will be  make it easier for the reader that we  give instructions with a list, and a title in bold for each step needed.

1) **Fork and Clone this Repository**
[Fork this repository](https://github.com/elixir-lang/elixir-lang.github.com/fork), and Create a local clone of your fork.  You may want to read the [Instructions on how to do that](https://help.github.com/articles/fork-a-repo/).

2) **Install [Ruby](https://www.ruby-lang.org/)**
To setup a development enviroment locally, you need to have Ruby installed. If you are running a Mac, it's probably installed. Run in your shell `ruby --version`.  Your Ruby version should begin with `1.9.3` or `2.0.0`. You can follow [this guide](https://www.ruby-lang.org/en/documentation/installation/) if you need to install it.

3) **Install  [Bundler](http://bundler.io/)**
```bash
gem install bundler
```

4) **Install Project Dependencies**
Once you have Bundler installed, `cd` into the forked project and download and install all the necessary dependencies by running:
```bash
$ bundle install
```
This will install `jekyll` and the `github-pages` package, which will install all the same packages with the same versions that are run in GitHub server, so we can test locally, as if we were [running live](http://elixir-lang.org).

5) **Run Jekyll**
Run a development server (with live-reloading on) on http://localhost:4000/
```bash
$ bundle exec jekyll serve
```

6) **Make your Changes and Test Them**
Now you're ready to make your changes! Be sure to test the changes locally using
the development server. Once you're done with your changes, [push those changes
to your fork](https://help.github.com/articles/pushing-to-a-remote/) and then [submit a **pull request**](https://help.github.com/articles/using-pull-requests/).

eksperimental commented:
you have read the source code here:
https://gist.github.com/eksperimental/a0079f0012ccac622368

eksperimental commented:
and we need have to remove the section saying "JavaScript processor", because that will be installed by Bundler

eksperimental commented:
Also, i think we can link to to [this sections of the elixir readme file](https://github.com/elixir-lang/elixir/blob/master/CONTRIBUTING.md#pull-requests) when we talk about forking and pull requests. because it's really clear.

whatyouhide commented:
Pushed some of the changes suggested by @eksperimental and added some markup niceties. I also removed the JavaScript preprocessor section and added the `gem install bundler` section. Thanks @eksperimental!

eksperimental commented on Gemfile:
```diff
@@ -0,0 +1,4 @@
+source 'https://rubygems.org'
+
+gem 'github-pages'
+gem 'jekyll-redirect-from'
```
I would personally delete this line, and let the github-pages package deal with it. Technically we are not supposed to install any plugins.

eksperimental commented on README.md:
```diff
@@ -24,22 +24,60 @@ export EDITOR=vim; _bin/newpost 'Post title'
+the development server. Once you're done with your changes, push those changes
+to your fork and then [submit a **pull
+request**](https://help.github.com/articles/using-pull-requests/). For a nice
+wrap-up on how to open a good pull request have a look at the [Elixir
```
that section has good instruction on "how to pull the latest project version from the server" the right way, so I think it's worth mentioning it.

eksperimental commented:
@whatyouhide it looks beautiful. I think it's very clear and much easier for anyone who wants to contribute now

josevalim commented:
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
